File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int[] temp = { -1, -1 };
        int n = nums.length;

        for (int i = 0; i < nums.length; i++) {
            int rem = target - nums[i];
            if (map.containsKey(rem)) {
                int idx = map.get(rem);
                temp[0] = idx;
                temp[1] = i;
                return temp;
            }
            map.put(nums[i], i);
        }

        return temp;
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        ans = [-1, -1]
        hashMap = {}
        for i in range(len(nums)):
            rem = target - nums[i]
            if rem in hashMap.keys():
                return [hashMap.get(rem), i]
            else:
                hashMap[nums[i]] = i

        return [-1, -1]
